functions/users/index.js

```diff
@@ -0,0 +1,102 @@
+        && data.hasOwnProperty('email')
+        && data.hasOwnProperty('password')
+        && data.hasOwnProperty('phone')
+        && data.hasOwnProperty('address')
+        && data.hasOwnProperty('postal_code')
+        && data.hasOwnProperty('country')
+        && data.hasOwnProperty('city')
+        && data.hasOwnProperty('date_of_birth')
+        && data.hasOwnProperty('gender');
+}
+
+exports.createUser = functions.https.onRequest((req, res) => {
+    if(req.method !== 'POST') {
+        return res.status(400).json({
+            message: 'Bad request, POST required'
+        })
+    }
+    else {
+        const data = req.body;
+        if (!checkUserData(data)) {
+            return res.status(400).json({
+                message: 'Missing parameters ' +
+                    'required: ' +
+                    'first_name, ' +
+                    'last_name, ' +
+                    'email, ' +
+                    'password, ' +
+                    'phone, ' +
+                    'address, ' +
+                    'postal_code, ' +
+                    'city, ' +
+                    'country, ' +
+                    'date_of_birth, ' +
+                    'gender'
+            })
+        }
+        admin.auth().createUser({
+            email: data.email,
+            emailVerified: false,
+            password: data.password,
+            disabled: false
+        })
+            .then((userRecord) => {
+                console.log("User created with UID of: " + userRecord.uid);
+                dbref.child('Users').child(userRecord.uid.toString()).set({
+                    first_name: data.first_name,
+                    last_name: data.last_name,
+                    email: data.email,
+                    phone: data.phone,
+                    address: data.address,
+                    postal_code: data.postal_code,
+                    city: data.city,
+                    country: data.country,
+                    date_of_birth: data.date_of_birth,
+                    gender: data.gender
+                }).then(() => {
+                    return res.status(200).json({
+                        message: 'User Created Successfully',
+                        email: userRecord.email,
+                        user_uid: userRecord.uid,
+                    });
+                })
+            })
+            .catch((error) => {
+                console.log("Error creating user: ", error);
+                return res.status(500).json({
+                    message: 'Error' + error.toString()
+                })
+            })
+    }
+});
+
+exports.updateUser = functions.https.onRequest((req, res) => {
+    if(req.method !== 'POST') {
+        return res.status(400).json({
+            message: "Bad request, POST Required"
+        })
+    } else {
+        const data = req.body;
+        dbref.child('Users').child(data.uid).update({
+            [data.updatedType] : data.value.toString()
```

Comment:
Yikes, this allows the user to modify all attributes? Shouldn't we restrict to specific ones?

Reply:
Yeah, good point. We only need to restrict email though.